backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/steps/Encode.kt

```diff
@@ -0,0 +1,31 @@
+package app.ehrenamtskarte.backend.stores.importer.steps
+
+import app.ehrenamtskarte.backend.stores.importer.PipelineStep
+import app.ehrenamtskarte.backend.stores.importer.types.AcceptingStore
+import org.apache.commons.text.StringEscapeUtils
+import org.slf4j.Logger
+
+class Encode(private val logger: Logger) : PipelineStep<List<AcceptingStore>, List<AcceptingStore>>() {
+
+    override fun execute(input: List<AcceptingStore>) = input.mapNotNull {
+        try {
+            return@mapNotNull it.copy(
+                name = it.name.decodeSpecialCharacters(),
+                discount = it.discount?.decodeSpecialCharacters()
+            )
+        } catch (e: Exception) {
+            logger.info("Exception while encoding $it for database import", e)
+        }
+        null
+    }
+
+    private fun String.decodeSpecialCharacters(): String {
+        // We often get a double encoded string, i.e. &amp;amp;
+        return StringEscapeUtils.unescapeHtml4(
+            StringEscapeUtils.unescapeHtml4(
+                this
+            )
+        ).replace("<br/>", "\n")
```

Comment:
What about tripple encoded string? :) We could change this to a while loop which applies unescapeHtml4 until the string no longer changes.

Possibly with a maximum iteration of 1000 to avoid hanging.

Reply:
Did not change this, just renamed the file. Please open a separate issue for this.

Reply:
Let's not think about triple encoded strings please :D

Reply:
Oke :D